File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.js
function solution(id_list, report, k) {
  const ans = new Array(id_list.length);
  ans.fill(0) 
  const report_list = {}
    
  id_list.forEach((user)=>{
      report_list[user] = []
  })
  
  report.forEach((user)=>{
      const [user_id, report_id] = user.split(' ')
      if(!report_list[report_id].includes(user_id)){
          report_list[report_id].push(user_id)
      }        
  })
  
  for(const key in report_list){
      if(report_list[key].length >= k){
          report_list[key].forEach((user)=>{
              ans[id_list.indexOf(user)] += 1
          })
      }
  }
  return ans;
}

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    reportnum = {i:0 for i in id_list}
    mailnum = {i:0 for i in id_list}
    report = list(set(report))
    
    for i in range(len(report)):
        report[i] = report[i].split()
        reportnum[report[i][1]] += 1
    
    reportedlist = [key for key, value in reportnum.items() if value>=k]
    for i in range(len(report)):
        if report[i][1] in reportedlist:
            mailnum[report[i][0]] += 1
        
    return list(mailnum.values())